woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -1350,3 +1355,42 @@ def test_accessor_set_index_errors(sample_df):
     error = "Index column must be unique"
     with pytest.raises(LookupError, match=error):
         sample_df.ww.set_index('age')
+
+
+def test_set_types(sample_df):
```

Comment:
Should we also test setting a semantic tag somewhere in this test?

Reply:
Makes sense. I'll update the checks to include one that sets semantic tags as well